pekko/src/main/scala/org/scanamo/ScanamoPekko.scala

```diff
@@ -46,7 +47,7 @@ object ScanamoPekko {
 }
 
 private[scanamo] object PekkoInstances {
-  implicit val monad: Monad[Source[*, NotUsed]] = new Monad[Source[*, NotUsed]] {
+  implicit val monad: Monad[Pekko] = new Monad[Pekko] {
```

Comment:
This fixes a compilation problem relating to [Scala 3's syntax for wildcard arguments in types](https://docs.scala-lang.org/scala3/reference/changed-features/wildcards.html):

```
[error] -- [E006] Not Found Error: /home/runner/work/scanamo/scanamo/pekko/src/main/scala/org/scanamo/ScanamoPekko.scala:49:35 
[error] 49 |  implicit val monad: Monad[Source[*, NotUsed]] = new Monad[Source[*, NotUsed]] {
[error]    |                                   ^
[error]    |                                   Not found: type *
```

The [definition for `Monad`](https://github.com/typelevel/cats/blob/4e976f97bba875419854c4ad4e3f87630b7d362c/core/src/main/scala/cats/Monad.scala#L33) is `Monad[F[_]]` - we're trying to say that `Monad` takes a type `F` that has a type-parameter, but we don't care what _that_ type parameter that is (hence it's represented by the `_` underscore). We're trying to say what the monadic container type is- eg `Future`, or `List`, but we don't care what it _contains_ (eg `Future[Int]`, or `List[String]`

In the migration to Scala 3, clashing representations for both wildcards and placeholders-for-type-parameters have used `_`, `*`, _and_ `?`, as clashes between new Scala 3 syntax and the the [kind projector](https://github.com/typelevel/kind-projector) compiler plugin have been resolved.

> * In Scala 3.0, * is available as a type parameter placeholder.
> * In Scala 3.2, * is deprecated in favor of _. A -rewrite option is available to rewrite one to the other.
> * In Scala 3.3, * is removed again, and all type parameter placeholders will be expressed with _.

In the Scanamo project, we use the `-Ykind-projector:underscores` parameter for `scalac`, which gives us [_"a migration path for users that want a one-time transition to syntax with _ as a type parameter placeholder"_](https://docs.scala-lang.org/scala3/reference/changed-features/wildcards.html#:~:text=a%20migration%20path,type%20parameter%20placeholder):

https://github.com/scanamo/scanamo/blob/3da03a973261f147c8668758ebfe6646b5f578f9/build.sbt#L79-L80

My _guess_ is that in the original Scanamo code (`Monad[Source[*, NotUsed]]`), `*` is being used as a placeholder-for-a-type-parameter - but in our Scala 3.3 world (with `-Ykind-projector:underscores`?), `*` isn't allowed to do that.

The fix takes advantage of the fact that Scala lets us miss out all the type-parameters anyhow, so long as...

https://github.com/scanamo/scanamo/blob/c232af6928d1ac0a5217be713d569b9442ceabbd/pekko/src/main/scala/org/scanamo/ScanamoPekko.scala#L43